diff --git a/parallel-rdp/interface.cpp b/parallel-rdp/interface.cpp
index 0b7ffd2..f685e77 100644
--- a/parallel-rdp/interface.cpp
+++ b/parallel-rdp/interface.cpp
@@ -119,6 +119,10 @@ static size_t fragment_size;
 static std::vector<bool> rdram_dirty;
 static uint64_t sync_signal;
 
+static bool guide_held = false;
+static bool r2_held = false;
+static bool fast_forward_toggled = false;
+
 static TTF_Font *message_font;
 static std::queue<Message> messages;
 
@@ -301,7 +305,117 @@ bool sdl_event_filter(void *userdata, SDL_Event *event) {
     wsi->init_surface_swapchain();
     wsi->begin_frame();
   }
+else if (event->type == SDL_EVENT_GAMEPAD_BUTTON_DOWN)
+{
+    switch (event->gbutton.button)
+    {
+    case SDL_GAMEPAD_BUTTON_GUIDE:
+        guide_held = true;
+        break;
+
+    case SDL_GAMEPAD_BUTTON_RIGHT_SHOULDER: // R1
+        if (guide_held)
+        {
+            callback.save_state = true;
+            return 1;
+        }
+        break;
+
+    case SDL_GAMEPAD_BUTTON_LEFT_SHOULDER: // L1
+        if (guide_held)
+        {
+            callback.load_state = true;
+            return 1;
+        }
+        break;
+
+    case SDL_GAMEPAD_BUTTON_WEST: // West
+        if (guide_held)
+        {
+            display_fps = !display_fps;
+            rdp_onscreen_message(
+                display_fps ? "FPS display ON" : "FPS display OFF",
+                MESSAGE_SHORT
+            );
+            return 1;
+        }
+        break;
+
+    case SDL_GAMEPAD_BUTTON_SOUTH: // South
+        if (guide_held)
+        {
+            callback.reset_game = true;
+            rdp_onscreen_message("Reset game", MESSAGE_SHORT);
+            return 1;
+        }
+        break;
+
+    case SDL_GAMEPAD_BUTTON_START: //Start
+        if (guide_held)
+        {
+            if (gfx_info.fullscreen)
+                callback.emu_running = false;
+            return 1;
+        }
+        break;
+
+    case SDL_GAMEPAD_BUTTON_MISC1: // Share
+        if (guide_held)
+        {
+          SDL_RunOnMainThread(ra_display_inprogress_achievements, nullptr, true);
+        }
+        break;
 
+    default:
+        if (guide_held)
+            return 1; // block other inputs while guide held
+        break;
+    }
+}
+else if (event->type == SDL_EVENT_GAMEPAD_AXIS_MOTION)
+{
+    switch (event->gaxis.axis)
+    {
+    case SDL_GAMEPAD_AXIS_RIGHT_TRIGGER: // R2
+    {
+        bool pressed = event->gaxis.value > 0;
+
+        if (pressed && !r2_held && guide_held)
+        {
+            fast_forward_toggled = !fast_forward_toggled;
+            callback.enable_speedlimiter = !fast_forward_toggled;
+
+            rdp_onscreen_message(
+                fast_forward_toggled ? "Fast forward on"
+                                     : "Fast forward off",
+                MESSAGE_SHORT
+            );
+
+            r2_held = true;
+            return 1;
+        }
+
+        if (!pressed)
+            r2_held = false;
+    }
+    break;
+
+    default:
+        break;
+    }
+}
+else if (event->type == SDL_EVENT_GAMEPAD_BUTTON_UP)
+{
+    switch (event->gbutton.button)
+    {
+    case SDL_GAMEPAD_BUTTON_GUIDE:
+        guide_held = false;
+        break;
+
+    default:
+        break;
+    }
+}
   return 0;
 }
 
@@ -428,9 +542,9 @@ void rdp_init(void *_window, GFX_INFO _gfx_info, const void *font,
   }
 
   message_font =
-      TTF_OpenFontIO(SDL_IOFromConstMem(font, font_size), true, 25.0);
+      TTF_OpenFontIO(SDL_IOFromConstMem(font, font_size), true, 30.0);
   achievement_challenge_indicator_font =
-      TTF_OpenFontIO(SDL_IOFromConstMem(font, font_size), true, 12.0);
+      TTF_OpenFontIO(SDL_IOFromConstMem(font, font_size), true, 30.0);
   if (!message_font || !achievement_challenge_indicator_font) {
     rdp_close();
     return;
diff --git a/src/ui/input.rs b/src/ui/input.rs
index 5cab30d..989a655 100644
--- a/src/ui/input.rs
+++ b/src/ui/input.rs
@@ -430,6 +430,19 @@ pub fn get(ui: &mut ui::Ui, channel: usize) -> InputData {
     let controller = ui.input.controllers[channel].game_controller;
     let joystick = ui.input.controllers[channel].joystick;
 
+    let guide_pressed = !controller.is_null()
+        && unsafe {
+            sdl3_sys::gamepad::SDL_GetGamepadButton(
+                controller,
+                sdl3_sys::gamepad::SDL_GamepadButton::GUIDE,
+            )
+        };
+    let game_controller = if guide_pressed {
+        std::ptr::null_mut()
+    } else {
+        controller
+    };
+
     let alt_pressed = unsafe {
         // ignore key presses if ALT is pressed
         *ui.input
@@ -444,12 +457,12 @@ pub fn get(ui: &mut ui::Ui, channel: usize) -> InputData {
     let mut keys = set_buttons(
         profile,
         joystick,
-        controller,
+        game_controller,
         ui.input.keyboard_state,
         alt_pressed,
     );
 
-    let (mut x, mut y) = set_axis(profile, joystick, controller, ui.input.keyboard_state);
+    let (mut x, mut y) = set_axis(profile, joystick, game_controller, ui.input.keyboard_state);
     bound_axis(&mut x, &mut y);
 
     keys |= (x.round() as i8 as u8 as u32) << X_AXIS_SHIFT;
