diff --git a/Cargo.toml b/Cargo.toml
index 683dbb3..27d7c04 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,7 @@
 name = "gopher64"
 version = "1.1.36"
 edition = "2024"
-rust-version = "1.98.0"
+rust-version = "1.94.1"
 
 [features]
 default = ["gui"]
diff --git a/rust-toolchain.toml b/rust-toolchain.toml
index b73c15e..9cf4a67 100644
--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.98.0"
+channel = "1.94.1"
