diff --git a/menu/drivers/ozone.c b/menu/drivers/ozone.c
--- a/menu/drivers/ozone.c
+++ b/menu/drivers/ozone.c
@@ -9824,7 +9824,7 @@ static void *ozone_init(void **userdata, bool video_is_threaded)
    ozone->last_framebuffer_opacity              = -1.0f;
    ozone->last_dims                             = out_dims;
    ozone->last_scale_factor                     = gfx_display_get_dpi_scale(p_disp,
-         settings, out_dims, false, false);
+         settings, out_dims, false, false) * 2.000f;
    ozone->last_thumbnail_scale_factor           = settings->floats.ozone_thumbnail_scale_factor;
 
    ozone->entries_old                           = NULL;
@@ -10852,7 +10852,7 @@ static void ozone_render(void *data,
    /* Check whether screen dimensions or menu scale
     * factor have changed */
    scale_factor               = gfx_display_get_dpi_scale(p_disp, settings,
-            dims, false, false);
+            dims, false, false) * 2.000f;
    thumbnail_scale_factor     = settings->floats.ozone_thumbnail_scale_factor;
    padding_factor             = settings->floats.ozone_padding_factor;
    font_scale_factor_global   = (font_scale == 1) ? (settings->floats.ozone_font_scale_factor_global) : 1.0f;
