diff --git a/menu/drivers/materialui.c b/menu/drivers/materialui.c
--- a/menu/drivers/materialui.c
+++ b/menu/drivers/materialui.c
@@ -4394,7 +4394,7 @@ static void materialui_render(void *data,
    /* Check whether screen dimensions, menu scale
     * factor or layout optimisation settings have changed */
    scale_factor = gfx_display_get_dpi_scale(p_disp, settings,
-         dims, false, false);
+         dims, false, false) * 2.000f;
 
    if (   (scale_factor != mui->last_scale_factor)
        || (dims != mui->last_dims)
@@ -9640,7 +9640,7 @@ static void *materialui_init(void **userdata, bool video_is_threaded)
    mui->last_dims                         = out_dims;
    mui->last_scale_factor                 = gfx_display_get_dpi_scale(
          p_disp, settings, out_dims,
-         false, false);
+         false, false) * 2.000f;
    mui->dip_base_unit_size                = mui->last_scale_factor
       * MUI_DIP_BASE_UNIT_SIZE;
    mui->flags                             = 0;
