diff --git a/gfx/gfx_widgets.c b/gfx/gfx_widgets.c
--- a/gfx/gfx_widgets.c
+++ b/gfx/gfx_widgets.c
@@ -1258,6 +1258,9 @@ static INLINE void gfx_widgets_update_layout(
             p_disp,
             settings, dims, fullscreen, true);
 
+   if (p_disp->menu_driver_id == MENU_DRIVER_ID_OZONE)
+      scale_factor = scale_factor * 1.500f;
+
    /* Check whether screen dimensions, menu scale factor or the
     * notification font have changed. The font is watched here rather
     * than from a settings callback because the rebuild needs the
