diff --git a/apps/ymir-sdl3/CMakeLists.txt b/apps/ymir-sdl3/CMakeLists.txt
index 4a1f8c10..f5362c6b 100644
--- a/apps/ymir-sdl3/CMakeLists.txt
+++ b/apps/ymir-sdl3/CMakeLists.txt
@@ -427,11 +427,9 @@ endif()
 find_package(cereal CONFIG REQUIRED)
 find_package(cxxopts CONFIG REQUIRED)
 find_package(date CONFIG REQUIRED)
-find_package(OpenSSL CONFIG REQUIRED)
 find_package(CURL REQUIRED)
 find_package(rtmidi CONFIG REQUIRED)
 find_package(SDL3 CONFIG REQUIRED)
-find_package(Stb REQUIRED)
 find_package(tomlplusplus CONFIG REQUIRED)
 
 target_include_directories(ymir-sdl3
@@ -453,7 +451,7 @@ target_link_libraries(ymir-sdl3 PRIVATE
     cereal::cereal
     lz4::lz4
     RtMidi::rtmidi
-    CURL::libcurl_static
+    CURL::libcurl
 )
 if (WIN32)
     target_link_libraries(ymir-sdl3 PRIVATE
diff --git a/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp b/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp
index 6ef1d0ec..d614c344 100644
--- a/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp
+++ b/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp
@@ -32,9 +32,6 @@
 #include <zlib.h>
 #include <zstd.h>
 
-#define MINIZ_NO_ZLIB_COMPATIBLE_NAMES
-#include <miniz/miniz.h>
-
 #define _STR_IMPL(x) #x
 #define _STR(x) _STR_IMPL(x)
 #define _SEMVER_STR(major, minor, patch) _STR(major.minor.patch)
@@ -49,7 +46,7 @@
 #define LZMA_VERSION "25.01"    // Private dependency of libchdr
 #define LIBCHDR_VERSION "0.3.0" // Not exported
 #define MIO_VERSION "1.1.0"     // Not exported
-// #define MZ_VERSION "3.1.1"      // Private dependency of libchdr
+#define MZ_VERSION "3.1.1"         // Private dependency of libchdr
 #define NLOHMANN_JSON_VERSION \
     _SEMVER_STR(NLOHMANN_JSON_VERSION_MAJOR, NLOHMANN_JSON_VERSION_MINOR, NLOHMANN_JSON_VERSION_PATCH)
 #define SDL_VERSION_STR _SEMVER_STR(SDL_MAJOR_VERSION, SDL_MINOR_VERSION, SDL_MICRO_VERSION)
diff --git a/vendor/libchdr/CMakeLists.txt b/vendor/libchdr/CMakeLists.txt
index bcf038cc..2ead7ff1 100644
--- a/vendor/libchdr/CMakeLists.txt
+++ b/vendor/libchdr/CMakeLists.txt
@@ -1,11 +1,5 @@
-if (Ymir_LIBRARY_ONLY)
-    add_subdirectory(libchdr/deps/miniz-3.1.1 EXCLUDE_FROM_ALL)
-    add_subdirectory(libchdr/deps/zstd-1.5.7 EXCLUDE_FROM_ALL)
-else ()
-    find_package(miniz CONFIG REQUIRED)
-    find_package(ZLIB REQUIRED)
-    find_package(zstd REQUIRED)
-endif ()
+add_subdirectory(libchdr/deps/miniz-3.1.1 EXCLUDE_FROM_ALL)
+add_subdirectory(libchdr/deps/zstd-1.5.7 EXCLUDE_FROM_ALL)
 add_subdirectory(libchdr/deps/lzma-25.01 EXCLUDE_FROM_ALL)
 
 list(APPEND CHDR_INCLUDES ${CMAKE_CURRENT_BINARY_DIR}/include)
@@ -28,11 +22,7 @@ add_library(chdr-static STATIC
 )
 target_include_directories(chdr-static PRIVATE ${CHDR_INCLUDES} PUBLIC libchdr/include)
 target_link_libraries(chdr-static PRIVATE chdr-lzma)
-if (Ymir_LIBRARY_ONLY)
-    target_link_libraries(chdr-static PRIVATE zstd miniz)
-else ()
-    target_link_libraries(chdr-static PRIVATE zstd::libzstd_static ZLIB::ZLIB miniz::miniz)
-endif ()
+target_link_libraries(chdr-static PRIVATE zstd miniz)
 if (MSVC)
     target_compile_definitions(chdr-static PRIVATE _CRT_SECURE_NO_WARNINGS)
 endif()
