Subject: [PATCH] arm64: dts: qcom: sm8550: add shallow CPU and CX retention idle states

--- a/arch/arm64/boot/dts/qcom/sm8550.dtsi
+++ b/arch/arm64/boot/dts/qcom/sm8550.dtsi
@@ -307,9 +307,47 @@
 				min-residency-us = <7480>;
 				local-timer-stop;
 			};
+
+			little_cpu_sleep_pc: cpu-sleep-0-1 {
+				compatible = "arm,idle-state";
+				idle-state-name = "silver-power-collapse";
+				arm,psci-suspend-param = <0x40000003>;
+				entry-latency-us = <250>;
+				exit-latency-us = <900>;
+				min-residency-us = <3200>;
+				local-timer-stop;
+			};
+
+			big_cpu_sleep_pc: cpu-sleep-1-1 {
+				compatible = "arm,idle-state";
+				idle-state-name = "gold-power-collapse";
+				arm,psci-suspend-param = <0x40000003>;
+				entry-latency-us = <400>;
+				exit-latency-us = <1100>;
+				min-residency-us = <4011>;
+				local-timer-stop;
+			};
+
+			prime_cpu_sleep_pc: cpu-sleep-2-1 {
+				compatible = "arm,idle-state";
+				idle-state-name = "goldplus-power-collapse";
+				arm,psci-suspend-param = <0x40000003>;
+				entry-latency-us = <450>;
+				exit-latency-us = <1200>;
+				min-residency-us = <6230>;
+				local-timer-stop;
+			};
 		};
 
 		domain-idle-states {
+			cluster_cx_ret: cluster-cx-ret {
+				compatible = "domain-idle-state";
+				arm,psci-suspend-param = <0x41001344>;
+				entry-latency-us = <1561>;
+				exit-latency-us = <2801>;
+				min-residency-us = <8550>;
+			};
+
 			cluster_sleep_0: cluster-sleep-0 {
 				compatible = "domain-idle-state";
 				arm,psci-suspend-param = <0x41000044>;
@@ -424,54 +462,54 @@
 		cpu_pd0: power-domain-cpu0 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&little_cpu_sleep_0>;
+			domain-idle-states = <&little_cpu_sleep_pc>, <&little_cpu_sleep_0>;
 		};
 
 		cpu_pd1: power-domain-cpu1 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&little_cpu_sleep_0>;
+			domain-idle-states = <&little_cpu_sleep_pc>, <&little_cpu_sleep_0>;
 		};
 
 		cpu_pd2: power-domain-cpu2 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&little_cpu_sleep_0>;
+			domain-idle-states = <&little_cpu_sleep_pc>, <&little_cpu_sleep_0>;
 		};
 
 		cpu_pd3: power-domain-cpu3 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&big_cpu_sleep_0>;
+			domain-idle-states = <&big_cpu_sleep_pc>, <&big_cpu_sleep_0>;
 		};
 
 		cpu_pd4: power-domain-cpu4 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&big_cpu_sleep_0>;
+			domain-idle-states = <&big_cpu_sleep_pc>, <&big_cpu_sleep_0>;
 		};
 
 		cpu_pd5: power-domain-cpu5 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&big_cpu_sleep_0>;
+			domain-idle-states = <&big_cpu_sleep_pc>, <&big_cpu_sleep_0>;
 		};
 
 		cpu_pd6: power-domain-cpu6 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&big_cpu_sleep_0>;
+			domain-idle-states = <&big_cpu_sleep_pc>, <&big_cpu_sleep_0>;
 		};
 
 		cpu_pd7: power-domain-cpu7 {
 			#power-domain-cells = <0>;
 			power-domains = <&cluster_pd>;
-			domain-idle-states = <&prime_cpu_sleep_0>;
+			domain-idle-states = <&prime_cpu_sleep_pc>, <&prime_cpu_sleep_0>;
 		};
 
 		cluster_pd: power-domain-cluster {
 			#power-domain-cells = <0>;
-			domain-idle-states = <&cluster_sleep_0>, <&cluster_sleep_1>;
+			domain-idle-states = <&cluster_cx_ret>, <&cluster_sleep_0>, <&cluster_sleep_1>;
 		};
 	};
 
